File: leetcode/287.find-the-duplicate-number.java
/*
 * @lc app=leetcode id=287 lang=java
 *
 * [287] Find the Duplicate Number
 */

// @lc code=start
class Solution {
    public int findDuplicate(int[] nums) {
        int res = -1;
        Arrays.sort(nums);
        for(int i=1; i<nums.length; ++i){
            if(nums[i]==nums[i-1]){
                res = nums[i];
                break;
            }
        }
        return res;
    }
}
// @lc code=end

File: leetcode/287.find-the-duplicate-number.cpp
/*
 * @lc app=leetcode id=287 lang=cpp
 *
 * [287] Find the Duplicate Number
 */

// @lc code=start
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int size = nums.size();
        int l = 1, h = size-1;
        while(l <= h) {
            int mid = l + (h-l)/2;
            int cnt = 0;
            for (size_t i = 0; i < size; i++){
                if(nums[i] <= mid) ++cnt;
            }
            if(cnt <= mid) l = mid + 1;
            else h = mid - 1;
        }
        return l;
    }
};
// @lc code=end

